src/main/java/appeng/me/cluster/implementations/CraftingCPUCluster.java

```diff
@@ -376,6 +384,23 @@ private void completeJob() {
         this.lastTime = 0;
         this.elapsedTime = 0;
         this.isComplete = true;
+
+        notifyRequester(false);
+        this.requestingPlayerUUID = null;
+    }
+
+    private void notifyRequester(boolean cancelled) {
+        if (!Platform.isServer()) return;
+        if (this.requestingPlayerUUID == null) return;
+        if (this.finalOutput == null) return;
+        if (!AEConfig.instance().isFeatureEnabled(AEFeature.CRAFTING_TOASTS)) return;
+
+        var player = PlayerHelper.getPlayerByUUID(this.requestingPlayerUUID);
+        if (player != null) {
+            try {
+                NetworkHandler.instance().sendTo(new PacketCraftingToast(this.finalOutput, cancelled), player);
```

Comment:
What happens here if on a server, the player logs off after queueing a craft but before it finishes? Does the packet just silently fail or does something worse happen?